exchanges/binance/binance_test.go

```diff
@@ -2938,3 +2938,234 @@ func TestGetUserMarginInterestHistory(t *testing.T) {
+	if _, err := b.CryptoLoanAdjustLTV(context.Background(), 42069, true, 1); err != nil {
+		t.Error(err)
+	}
+}
+
+func TestCryptoLoanLTVAdjustmentHistory(t *testing.T) {
+	t.Parallel()
+	sharedtestvalues.SkipTestIfCredentialsUnset(t, b)
+	if _, err := b.CryptoLoanLTVAdjustmentHistory(context.Background(), 0, currency.USDT, currency.BTC, time.Time{}, time.Time{}, 0, 0); err != nil {
+		t.Error(err)
+	}
+}
+
+func TestCryptoLoanAssetsData(t *testing.T) {
+	t.Parallel()
+	sharedtestvalues.SkipTestIfCredentialsUnset(t, b)
+	if _, err := b.CryptoLoanAssetsData(context.Background(), currency.EMPTYCODE, 0); err != nil {
+		t.Error(err)
+	}
+}
+
+func TestCryptoLoanCollateralAssetsData(t *testing.T) {
+	t.Parallel()
+	sharedtestvalues.SkipTestIfCredentialsUnset(t, b)
+	if _, err := b.CryptoLoanCollateralAssetsData(context.Background(), currency.EMPTYCODE, 0); err != nil {
+		t.Error(err)
+	}
+}
+
+func TestCryptoLoanCheckCollateralRepayRate(t *testing.T) {
+	t.Parallel()
+	if _, err := b.CryptoLoanCheckCollateralRepayRate(context.Background(), currency.EMPTYCODE, currency.BNB, 69); !errors.Is(err, errLoanCoinMustBeSet) {
+		t.Errorf("received %v, expected %v", err, errLoanCoinMustBeSet)
+	}
+	if _, err := b.CryptoLoanCheckCollateralRepayRate(context.Background(), currency.BUSD, currency.EMPTYCODE, 69); !errors.Is(err, errCollateralCoinMustBeSet) {
+		t.Errorf("received %v, expected %v", err, errCollateralCoinMustBeSet)
+	}
+	if _, err := b.CryptoLoanCheckCollateralRepayRate(context.Background(), currency.BUSD, currency.BNB, 0); !errors.Is(err, errAmountMustBeSet) {
+		t.Errorf("received %v, expected %v", err, errAmountMustBeSet)
+	}
+
+	sharedtestvalues.SkipTestIfCredentialsUnset(t, b)
+	if _, err := b.CryptoLoanCheckCollateralRepayRate(context.Background(), currency.BUSD, currency.BNB, 69); err != nil {
+		t.Error(err)
+	}
+}
+
+func TestCryptoLoanCustomiseMarginCall(t *testing.T) {
+	t.Parallel()
+	if _, err := b.CryptoLoanCustomiseMarginCall(context.Background(), 0, currency.BTC, 0); err == nil {
+		t.Error("expected an error")
+	}
+
+	sharedtestvalues.SkipTestIfCredentialsUnset(t, b, canManipulateRealOrders)
+	if _, err := b.CryptoLoanCustomiseMarginCall(context.Background(), 1337, currency.BTC, .70); err != nil {
+		t.Error(err)
+	}
+}
+
+func TestFlexibleLoanBorrow(t *testing.T) {
+	t.Parallel()
+	if _, err := b.FlexibleLoanBorrow(context.Background(), currency.EMPTYCODE, currency.USDC, 1, 0); !errors.Is(err, errLoanCoinMustBeSet) {
+		t.Errorf("received %v, expected %v", err, errLoanCoinMustBeSet)
+	}
+	if _, err := b.FlexibleLoanBorrow(context.Background(), currency.ATOM, currency.EMPTYCODE, 1, 0); !errors.Is(err, errCollateralCoinMustBeSet) {
+		t.Errorf("received %v, expected %v", err, errCollateralCoinMustBeSet)
+	}
+
+	sharedtestvalues.SkipTestIfCredentialsUnset(t, b, canManipulateRealOrders)
+	if _, err := b.FlexibleLoanBorrow(context.Background(), currency.ATOM, currency.USDC, 1, 0); err != nil {
+		t.Error(err)
```

Comment:
`{"code":-9000,"msg":"An unknown error occurred while processing the request."}`
It may not be an issue given locality restrictions, but this error stands out from the others

Reply:
Looks to be related to restrictions but without a helpful error:

Bad:
`{"code":-10044,"msg":"Collateral amount has reached the limit. Please reduce your collateral amount or try with other collaterals."}`

Good:
`{"collateralCoin":"BTC","loanCoin":"USDC","loanAmount":"69","collateralAmount":"420","status":"Succeeds"}`